gnovm/pkg/gnolang/preprocess.go

```diff
@@ -1738,7 +1738,12 @@ func Preprocess(store Store, ctx BlockNode, n Node) Node {
 			case *KeyValueExpr:
 				// NOTE: For simplicity we just
 				// use the *CompositeLitExpr.
-
+			// TRANS_LEAVE -----------------------
+			case *StarExpr:
+				xt := evalStaticTypeOf(store, last, n.X)
+				if xt.Kind() != PointerKind && xt.Kind() != TypeKind {
+					panic(fmt.Sprintf("invalid operation: cannot indirect %s (variable of type %s)", n.X.String(), xt.String()))
```

Comment:
consider this one:
```go
package main

func main() {
	println(*nil)
}
```
but yes I'll make a new PR.

Reply:
always ahead of the game :) should have let you take a look first

Reply:
nice catch. Missed this one ^^

Reply:
did a fix here: https://github.com/gnolang/gno/pull/3053